### ISSUE TITLE
[v2][Feat] Advanced options

### ISSUE THREAD

**Issue by chimera-defi** [SharedStake/SharedStake-ui#155]
Add a advanced options popups similar to sushi-swap which allows the user to pick:
- referral address
- validator software they want e.g. prysm
-

**Comment by chimera-defi** [comment]
note

So I'm thinking of todos cos this a relatively large feature that i didnt fully scope out. 

1. 
We also want users that didnt enter the site with the ref param to be able to manually enter their ref addr and/or other params. 
See https://github.com/SharedStake/SharedStake-ui/issues/155

Basically re-use the dropdowns on landing page with a dropdown to add options into a contract call. 
As a re-useable component. Used in the staking page at least for now. 

This is the WIP contract for ref.
https://github.com/chimera-defi/SharedDeposit/blob/main/contracts/v2/periphery/UserDepositHelper.sol

Basically just emits events and forwards the eth to the minter and tokens back to the user. 
BUT we do need to check what works best for calculating an airdrop. just some js code to check the events onchain. can even use airstack maybe. 
rn its 4 events, each named, with the user, total value, and input.

**Comment by t-e-blake** [comment]
Understood, this will be easy to implement with a simple (pseudo) `if (refAddress): refInputValue=refAddress` 

Is this to be added to only the staking page?
